ladybug/sunpath.py

```diff
@@ -117,38 +154,47 @@ def is_leap_year(self, value):
 
     @property
     def daylight_saving_period(self):
-        """Get and set daylight saving period.
+        """Get or set an AnalysisPeriod for the daylight saving period.
+
+        Note that the st_hour and end_hour of the AnalysisPeriod will be interpreted
+        as indicating the time of day when the shift in clocks happen. So setting the
+        st_hour and end_hour to 2 will ensure that the time shift happens at 2 AM on
+        the dates specified. Note that this is different than how the hour inputs of
+        AnaylsisPeriods typically work, where the st_hour and end_hour apply to every
+        day of the analysis period.
 
-        daylight saving period should be an AnalysisPeriod.
+        If None, no daylight saving period will be used.
         """
         return self._daylight_saving_period
 
     @daylight_saving_period.setter
     def daylight_saving_period(self, value):
-        if value:
+        if value is not None:
             assert isinstance(value, AnalysisPeriod), \
                 'Daylight saving period should be an AnalysisPeriod not %s' % type(value)
-
         self._daylight_saving_period = value
 
     def is_daylight_saving_hour(self, datetime):
-        """Check if a datetime is a daylight saving time."""
+        """Check if a datetime is within the daylight saving time."""
         if not self.daylight_saving_period:
             return False
-        return self.daylight_saving_period.is_time_included(datetime)
+        return self.daylight_saving_period.st_time.moy <= datetime.moy < \
+            self.daylight_saving_period.end_time.moy
 
     def calculate_sun(self, month, day, hour, is_solar_time=False):
         """Get Sun data for an hour of the year.
 
         Args:
-            month: An integer between 1-12
-            day: An integer between 1-31
-            hour: A positive number between 0..23
+            month: An integer between 1 and 12.
+            day: An integer between 1 and 31.
+            hour: A positive number between 0 and 23.
             is_solar_time: A boolean to indicate if the input hour is solar time.
-                (Default: False)
+                Note that this input is redundant and should not be used if the
+                time_zone of the sun path has already been set to a decimal
+                solar time. (Default: False)
```

Comment:
Can we think of a better way to implement this?

Reply:
I think it's useful to have the two different ways to get solar time:
1. Set this argument to True
2. Set the time zone of the sun path to be `None`.
The first one means that you can still get the solar time even if you init a Sunpath from an EPW location with an integer time zone and the second one is useful for cases where you don't know/don't care about time zones and just need some solar positions that reflect the latitude.

I think I may just remove this part of the docstring, though, since it's not really helpful or correct. It's fine if people set it to True while using a `Sunpath.time_zone` that's set to solar time. It's an extra step that won't change the result but it' not like it breaks or it's incorrect if they set it to True for this case.

Reply:
Updated the dcocstring.